apps/studio/components/interfaces/Functions/EdgeFunctionSecrets/EditSecretModal.tsx

```diff
@@ -0,0 +1,211 @@
+import { useEffect, useState } from "react"
+import { SubmitHandler, useForm } from "react-hook-form"
+import { zodResolver } from "@hookform/resolvers/zod";
+import { toast } from 'sonner'
+import z from "zod";
+
+import { Eye, EyeOff, X } from "lucide-react";
+import { useParams } from 'common'
+import {
+  Button,
+  Form_Shadcn_,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Input,
+  Input_Shadcn_,
+  Separator,
+  Sheet,
+  SheetClose,
+  SheetContent,
+  SheetFooter,
+  SheetHeader,
+  SheetSection,
+  SheetTitle,
+  cn
+} from "ui";
+import ConfirmationModal from "ui-patterns/Dialogs/ConfirmationModal";
+import { FormItemLayout } from "ui-patterns/form/FormItemLayout/FormItemLayout";
+import { useSecretsCreateMutation } from 'data/secrets/secrets-create-mutation'
+import { ProjectSecret } from "data/secrets/secrets-query";
+
+const FORM_ID = 'edit-secret-sidepanel'
+
+const FormSchema = z.object({
+  value: z.string().min(1, 'Please provide a value for your secret'),
+});
+
+type FormSchemaType = z.infer<typeof FormSchema>;
+
+interface EditSecretModalProps {
+  secret?: ProjectSecret
+  visible: boolean
+  onClose: () => void
+}
+
+export function EditSecretModal({ secret, visible, onClose }: EditSecretModalProps) {
+  const { ref: projectRef } = useParams()
+
+  const [isClosingPanel, setIsClosingPanel] = useState(false)
+  const [focusedEditor, setFocusedEditor] = useState(false)
+  const [showSecretValue, setShowSecretValue] = useState(false)
+
+  const form = useForm<FormSchemaType>({
+    resolver: zodResolver(FormSchema)
+  })
+
+  const { mutate: updateSecret, isLoading: isUpdating } = useSecretsCreateMutation({
+    onSuccess: (_, variables) => {
+      toast.success(`Successfully updated secret "${variables.secrets[0].name}"`)
+      onClose()
+    },
+  })
+
+  function isClosingSidePanel() {
+    form.formState.isDirty ? setIsClosingPanel(true) : onClose()
+  }
+  const onSubmit: SubmitHandler<FormSchemaType> = async (data) => {
+    if (secret === undefined) {
+      return;
+    }
+
+    updateSecret({
+      projectRef, secrets: [
+        { name: secret?.name, value: data.value }
+      ]
+    })
+  }
+
+  useEffect(() => {
+    if (visible) {
+      setFocusedEditor(false)
+      form.reset({ value: '' })
+    }
+  }, [visible, secret])
+
+  return (
+    <Sheet open={visible} onOpenChange={() => isClosingSidePanel()}>
```

Comment:
Let's rename this file to `EditSecretSheet.tsx` to keep in line with our other sheets.